### PR TITLE
0037 sudoku solver

### PR DESCRIPTION
**Pull request by vigneshshiv**
Tried best to explain, let me know if any changes needed.

### CODE REVIEW

File: solutions/0000-0099/0037-sudoku-solver-hard.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+As we all would have try to solve Sudoku problem in the past, it's `9 x 9` grid with numbers. 
+
+As stated above, a number (in cell) should appear only once in row and column and `3 x 3` block. 
+
+Let's walk through with the given example above, 1st row - `"5","3",".",".","7",".",".",".","."`. The 3rd column (2nd Index) is not filled with any fith any number. 
+
+So what are the possibilities 3rd column can have? Well, It can be from `1-9`, So we use 3 types of checks to place a number on the board. 
+
+1. Check the current row, columns from `1-9`.
+2. Check the current column, rows from `1-9`.
```
Comment (wingkwong) [inline solutions/0000-0099/0037-sudoku-solver-hard.md:76]: `1-9` -> `1 - 9`

File: solutions/0000-0099/0037-sudoku-solver-hard.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+So for each cell, we trying with $n$ possibilities, The process continues till for all the rows and columns. If some cases we might end up being not able to place any number and we reached last number that is $9$. **So it means we have placed some wrong order in the past, it requires us to validate the previous placed numbers on the board to solve the current cell problem.** 
+
+This types of situation problems can be solved with backtracking technique. If any of the current operation or action is going to cause future actions then we need to backtrack the previous cases, like placing next number and next number so on to solve the current problem. 
+
+At the end, we would have solved all the cells in the board. 
+
+Time complexity: O(9^n^2) - Exponential time complexity. 
+
+Space complexity: $O(n)$
```
Comment (wingkwong) [inline solutions/0000-0099/0037-sudoku-solver-hard.md:95]: Can you double check or explain how to get this.

Comment (vigneshshiv) [reply solutions/0000-0099/0037-sudoku-solver-hard.md:95]: We have total 9 numbers, for any number, n^2 is the possibilities. For every single cell, we are trying out n^2 possibilities, so it's 9^n^2 time complexity. 

Reference - https://www.geeksforgeeks.org/sudoku-backtracking-7/

File: solutions/0000-0099/0037-sudoku-solver-hard.md (hunk 1 above)
Comment (vigneshshiv) [reply solutions/0000-0099/0037-sudoku-solver-hard.md:76]: Yeah, will change it

**Comment by vigneshshiv** [comment]
Updated the number syntax, check once.

File: solutions/0000-0099/0037-sudoku-solver-hard.md (hunk 2 above)
Comment (wingkwong) [reply solutions/0000-0099/0037-sudoku-solver-hard.md:95]: Why SC is O(n)?

Comment (vigneshshiv) [reply solutions/0000-0099/0037-sudoku-solver-hard.md:95]: It's 2D grid, it's a perfect square of 9 x 9, so it's n^2 space complexity. 

Here, m == n, where m - # of rows, n - # of columns.

**Pull request merged**